### ISSUE TITLE
MA0056 should be marked as similar to CA2214

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1023]
### Version of the Meziantou.Analyzer NuGet package

3.0.10

### Rule Identifier

MA0056

### Target Framework

.NET 10.0

### C# Language version

C# 14

### Description

MA0056 is currently documented as being an equivalent of [CA2214](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2214?WT.mc_id=DT-MVP-5003978), but - unless I'm mistaken - that does short of *MA0056* as:
* CA2214 (nor S1699) is not reported when invoking virtual properties.
* CA2214 is not reported when invoking virtual methods defined on a base class.

Let me know if you want a PR for this or if copilot is up for the job.

PS. The name of the class and the documentation (except for the title) makes you think the rules only applies to method.

### Reproduction Steps

Minimal code:

```c#
public class A
{
    public A()
    {
        RunA(); // MA0056 / CA2214 / S1699
        System.Console.WriteLine(NameA); // MA0056
    }

    public virtual void RunA()
    {
    }

    public virtual string NameA
    {
        get
        {
            return string.Empty;
        }
    }
}

public class B : A
{
    public B()
    {
        RunA(); // MA0056 / S1699
        System.Console.WriteLine(NameA); // MA0056

        RunB(); // MA0056 / CA2214 / S1699
        System.Console.WriteLine(NameB); // MA0056
    }

    public virtual void RunB()
    {
    }

    public virtual string NameB
    {
        get
        {
            return string.Empty;
        }
    }
}
```
